### PR TITLE
Rename `Nx.power` to `Nx.pow`

### PR DESCRIPTION
**Pull request by Benjamin-Philip**
Closes https://github.com/elixir-nx/nx/issues/982.

### CODE REVIEW

File: exla/mix.exs [hunk 1]
```diff
@@ -59,8 +59,8 @@ defmodule EXLA.MixProject do
 
   defp deps do
     [
-      {:nx, "~> 0.4.2"},
-      # {:nx, path: "../nx"},
+      # {:nx, "~> 0.4.2"},
+      {:nx, path: "../nx"},
```
Comment (Benjamin-Philip) [inline exla/mix.exs:63]: I had to make this change to test. How do you compare and test EXLA with Nx main?

File: nx/lib/nx.ex [hunk 2]
```diff
@@ -3866,7 +3866,11 @@ defmodule Nx do
 
   """
   @doc type: :element
-  def power(left, right), do: element_wise_bin_op(left, right, :power, & &1)
+  def pow(left, right), do: element_wise_bin_op(left, right, :pow, & &1)
+
+  @deprecated "Use pow/2 instead"
+  @doc type: :element
+  def power(left, right), do: pow(left, right)
```
Comment (Benjamin-Philip) [inline nx/lib/nx.ex:3873]: How does this deprecation look?

File: exla/mix.exs (hunk 1 above)
Comment (josevalim) [reply exla/mix.exs:63]: By doing exactly this. :)

File: nx/lib/nx.ex [hunk 3]
```diff
@@ -3866,7 +3866,11 @@ defmodule Nx do
 
   """
   @doc type: :element
-  def power(left, right), do: element_wise_bin_op(left, right, :power, & &1)
+  def pow(left, right), do: element_wise_bin_op(left, right, :pow, & &1)
+
+  @deprecated "Use pow/2 instead"
+  @doc type: :element
```
Comment (josevalim) [inline nx/lib/nx.ex:3872]:
```suggestion
  @doc false
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**